http/exchange_api.go

```
@@ -0,0 +1,30 @@
+package http
+
+import (
+	"github.com/gin-gonic/gin"
+
+	"github.com/KyberNetwork/reserve-data/common"
+	"github.com/KyberNetwork/reserve-data/http/httputil"
+)
+
+// CancelAllOrders cancel all orders
+func (s *Server) CancelAllOrders(c *gin.Context) {
+	pendingActivites, err := s.app.GetPendingActivities()
+	if err != nil {
+		httputil.ResponseFailure(c, httputil.WithError(err))
+		return
+	}
+
+	for _, activity := range pendingActivites {
+		if activity.Action == common.ActionTrade {
+			exchange := activity.Params["exchange"].(common.Exchange)
+			// Cancel order
+			if err := s.core.CancelOrder(activity.ID, exchange); err != nil {
```

if an activity status get changed during we process, this will abort all the next one, we can add a convenient param like 'on_error == continue' to ignore error?, and we can have result as a list of which success/fail. maybe we should ask Luke if he need this.

---

Yes, I think about this before. Let me ask him if there is an error, we stop or continue cancel other orders. Maybe we should cancel as many order as we can.

---

Updated